app/controllers/admin_applications_controller.rb

```diff
@@ -2,10 +2,16 @@ class AdminApplicationsController < ApplicationController
   def show
     @applicant = Applicant.find(params[:id])
     @pets = @applicant.associated_pets(params[:id])
-    if params[:pet_id].nil?
-
-    else
-      @pet_id = (params[:pet_id])
+    @pet_applications = {}
+    @pets.each do |pet|
+      @pet_applications[pet.id] = PetApplicant.find_by_parents(pet.id, params[:id]).status
```

Comment:
move this to a private helper method so that in the action it just looks like
` @pet_applications = find_all_applications` where `find_all_applications` is a helper method that hold the logic below.